### ISSUE TITLE
Systray account state indicators: What states to show here?

### ISSUE THREAD

**Issue by DominiqueFuchs** [nextcloud/desktop#2210]
#2206 showed possible confusion (besides of an actual bug that is already handled in #2209) about the account state indicators: these are currently only distinguishing account connection itself (OK/NOK). It may be assumed that these also show the pause state.

IMHO: This currently wouldn't make sense to implement b/c it would only be useful in addition to the systray icon itself if sync-pause can be activated/deactivated per account. Right now you can only pause and resume all accounts, e.g. the whole client.

Opinions welcome (@jancborchardt ?)

**Comment by r2evans** [comment]
I certainly see utility in pausing one account and keeping another account(s) active and syncing. The drop-down icons should show the state of that account: green (okay), yellow (some warning/note), gray (paused?), red (failure). The systray icon likely shows the first match from among:

- any account shows red? show red
- any account shows yellow? show yellow
- any account paused? show s similar gray icon, but only a gray "paused" button if *all* accounts paused
- all accounts green? show green

I think simple is better here: if the colors are ordered, then show the "max" (or "min", depending on your perspective) of the colors of the underlying accounts. The only fuzzy thing is gray, where I wonder if the gray "paused" systray should only be shown if all are paused. Perhaps if there are a mix of green and paused, can you create a half-green/half-gray icon?

**Comment by jancborchardt** [comment]
Totally agree with @r2evans list of logic there. Any warnings or states except "all accounts green" should bubble up and also override the overall icon.

If there is any error in any of the accounts, the 1 tray icon indicator should be an ⓧ. When the error is with an account which is not the current one, we can show the ⓧ / other status indicator together with the ▼ so people know it’s a different account? Opening the account switcher will then indicate which account it is.

**Comment by r2evans** [comment]
One thought for a hybrid icon:

![image](https://user-images.githubusercontent.com/5815808/87948247-4ebb0180-ca59-11ea-83b3-671c9f22afac.png)

(I reserve no rights to this design, I leave licensing to Nextcloud designers.)

**Comment by jospoortvliet** [comment]
> One thought for a hybrid icon:
> 
> ![image](https://user-images.githubusercontent.com/5815808/87948247-4ebb0180-ca59-11ea-83b3-671c9f22afac.png)
> 
> (I reserve no rights to this design, I leave licensing to Nextcloud designers.)

It's a clever idea for sure, but the issue is that you can have a LOT of different statusses already with 2 accounts and what if you have, like, 5 or 6? In the end, the goal of the icon is to give an indication of the state, not to be the end and all of it, so I agree with Jan and @r2evans on this one.

**Comment by jancborchardt** [comment]
Yep – the bubbling up is simplest, and on top of it we don’t have enough space in tray icons to show a split icon like this, it will be confusing. It doesn’t even make clear _which_ of the accounts is synced and which is paused. Hence just a full pause icon works just as well.

**Comment by r2evans** [comment]
@jancborchardt, my point was that the hybrid indicates mismatch exists. It is *not* intended to:

- show *which* account is paused; or
- show *how many* accounts are paused/running.

In configurations where there is only one account, this icon will never be used.

To address your *"don't have enough space in tray icons"*, here are some more examples. The top is my current systray, the remaining are concepts:

![image](https://user-images.githubusercontent.com/5815808/89545463-3c122d80-d7b8-11ea-840a-4020ef6b803b.png)

(Again, I reserve no rights to these designs, I leave licensing to Nextcloud designers.)

I suggest that even if it is unclear which account is in which state, I'd think opening up the systray popup menu and seeing which account is not green would clear that up.

Except that's the original issue: despite me pausing one of my accounts, I see this (on 2.7.0.15641):

![image](https://user-images.githubusercontent.com/5815808/89545654-7da2d880-d7b8-11ea-804f-54f9a4524583.png)

The current state of the whole system (both accounts) and the state of one of the accounts ... is just wrong. On the original issue, I can't rely on icons and therefore have to resort to opening the Settings window and cycle through accounts to even know *if* something is paused. What's more confusing ... one more icon differentiating between "all good" and "at least one paused", or ... green icons that may mean not-green?

We can't completely trust the icons (systray and individual account) as-is, so I suggest that something has to change.

**Comment by mgallien** [comment]
@jancborchardt 
@jospoortvliet 
should we pursue work on this here or should I close it because we want to do a more general design review on the desktop client ?

**Comment by FlexW** [comment]
I will close that ticket for now because there seems to be no progress. Feel free to open it again.

**Comment by r2evans** [comment]
Close it because management does not expect to fulfill this feature request (with an explanation, please). Close it because it conflicts with something else (a reference to the conflict is always nice). But when the company has 500+ issues in this repo and 1000s across just a few of its other major repos, low-volume on some FR issues is to be expected.

Perhaps it would be better to ping those who were previously asked for input and have not yet replied.
@jospoortvliet 
@jancborchardt

**Comment by jancborchardt** [comment]
My comment still stands: https://github.com/nextcloud/desktop/issues/2210#issuecomment-661018180

It should bubble up, no hybrid icon.

**Comment by FlexW** [comment]
Thanks, @jancborchardt @r2evans for the feedback. Is there then anything left to do? From my understanding, everything works as expected.

**Comment by r2evans** [comment]
What do you mean by *"works as expected'*? If you think that the above discussion about mixed-status icons is somehow implemented, then I'm really missing something. Currently on the 3.3.4 desktop client (win10), I have paused one connection and kept the other active, and the main icon is full-green (as if all connections were enabled), and the one that is paused is showing active in the tray popup (but not in the "Settings" window). Only when I pause both (of two) connections will the tray show "paused".

Bottom line, the "bubbling up" (with or without a hybrid icon) is not happening.

@FlexW, how am I misinterpreting your statement and/or the state of the current behavior?

**Comment by FlexW** [comment]
@r2evans I guess the hybrid icon approach will never be implemented because of the reasons that @jancborchardt and @jospoortvliet gave. For the issue that the pause state (and maybe others) are not bubbling up, I  created another issue to keep the issue focused. See https://github.com/nextcloud/desktop/issues/3843

**Comment by claell** [comment]
Marking as duplicate of #619

**Comment by claell** [comment]
Duplicate of #619

**Comment by claell** [comment]
Apparently, #619 has nothing to do with using multiple accounts.

**Comment by r2evans** [comment]
@claell you closed it as a dupe then unmarked it (no longer a dupe) but are going to keep it closed?

**Comment by r2evans** [comment]
Missed that. I didn't realize that FlexW was a nextcloud employee and/or allowed to close other peoples' issues, now I see them on the Contributors page.

I'm repeatedly frustrated by the indicators on a multi-account installation, too bad NC doesn't consider the confusion a concern.

Thanks for replying.

**Comment by claell** [comment]
I agree there is room for improvement and originally came to this issue from the issue that I recently created: #4152. Maybe that can be used as a meta issue for the UX of the tray with multiple accounts (not only the icon has room for improvement).

**Comment by mgallien** [comment]
@claell if you close the issue by error, please reopen it

**Comment by claell** [comment]
@mgallien As I already wrote, I did not close it by error. This has already been closed originally by @FlexW before:
![grafik](https://user-images.githubusercontent.com/6171135/149163214-c555c337-351f-4139-9c29-261b44e9ffb8.png)

So I'd assume it should also stay closed, then?

**Comment by r2evans** [comment]
I find the premise *"seems to be no progress"* does not provide sufficient rationale to close an issue. There are issues in nextcloud/desktop over 4 years old (and over 5.5y in nextcloud/server) that have shown similarly negligible effort yet they remain open.

If it needs to be closed, close it for a something less intangible:

- *"while a good idea, the company cannot direct resources at this time"* (this feels weak but is a business decision that one cannot argue unless a paying client ... and I'm assuming that the business decision was made by somebody with some authority)
- *"we disagree with the premise that this will add sufficient value"* (I hope there was an attempt to clarify the problems)
- *"this affects too few users to warrant attention at this time"* (please find a way to quantify this)
- *"there is no clear path forward to fix this bug/feature, all offered solutions either do not resolve enough or make something worse"* (this assumes frequent and/or contentious discussion)

I don't know @FlexW and mean no disrespect, but I see nothing on their [profile](https://github.com/FlexW) that suggests they are a Nextcloud employee or manager, not even "member of Nextcloud". (For that matter, same with @claell). The presence of [commits by this user](https://github.com/nextcloud/desktop/commits?author=FlexW) does not mean that they are positioned to make these decisions. (Again, please I mean no disrespect by any of this. I'm just a user, wondering how "close" decisions are made and by whom. The fact that the owner of the repo has permitted somebody to close an issue may suffice from a public-perception view.)

Thanks.

**Comment by claell** [comment]
Regarding the argument "no progress", I agree that this should not be a reason to close.

To the other points: To my knowledge, @FlexW was previously employed at Nextcloud and that was also the time when he closed the issue.
I am not an employee of Nextcloud and try to be very conservative when it comes to closing issues.

Since the decision has been made by a former employee, I don't want to interfere with that. That is up to the current Nextcloud representatives, IMHO.

**Comment by r2evans** [comment]
Thanks @claell, that makes sense, I hadn't considered that. I think I need to be more aware of that possibility in this context, thank you for the perspective.

**Comment by Rello** [comment]
Hello,

there are different implications of account specific status in the task bar.
Also there are several dependencies as there is also a branded client for enterprise customers.
And there are the dependencies of the different task bars of windows and macOS (later one not even having colors).

We are looking into the topic and might rework the icons to transition to a Nextcloud application logo for example. But there is not timeframe on this. I am closing this bug report for now

**Comment by r2evans** [comment]
This makes little sense from a technical perspective.

- "branded clients" ... okay, so make the branding deal with icons and half-icons. It's added effort on those doing the branding, but they get to control it. It's not a lot more work.  You can even default the "half-icon" sets to be identical to the full (i.e., status quo from now) so that they don't have to do the extra work, that's a reasonable policy.
- "not even having colors" ... is the same as "branded clients".
- "dependencies of the different task bars" ... so you already have the deps _met_ in order to show the icon, but you cannot change the icon-set without affecting the dependencies?

If NC feels this request is unjustified or the programmers are overcome by other priorities, please just say that, it's been safe to infer this anyway based on the lack of action over the last 5 years. Please either provide more information that justifies closing this on **technical merit**, or be candid about the workload/priority/interest deficiencies.
